benchmarks/Microsoft.AspNetCore.SignalR.Microbenchmarks/RedisProtocolBenchmark.cs

```diff
@@ -0,0 +1,148 @@
+            _writtenInvocationSmallExclusions = _protocol.WriteInvocation(_methodName, _args, _excludedIdsSmall);
+            _writtenInvocationLargeExclusions = _protocol.WriteInvocation(_methodName, _args, _excludedIdsLarge);
+        }
+
+        [Benchmark]
+        public void WriteAck()
+        {
+            _protocol.WriteAck(42);
+        }
+
+        [Benchmark]
+        public void WriteGroupCommand()
+        {
+            _protocol.WriteGroupCommand(_groupCommand);
+        }
+
+        [Benchmark]
+        public void WriteInvocationNoExclusions()
+        {
+            _protocol.WriteInvocation(_methodName, _args);
+        }
+
+        [Benchmark]
+        public void WriteInvocationSmallExclusions()
+        {
+            _protocol.WriteInvocation(_methodName, _args, _excludedIdsSmall);
+        }
+
+        [Benchmark]
+        public void WriteInvocationLargeExclusions()
+        {
+            _protocol.WriteInvocation(_methodName, _args, _excludedIdsLarge);
+        }
+
+        [Benchmark]
+        public void ReadAck()
+        {
+            _protocol.ReadAck(_writtenAck);
+        }
+
+        [Benchmark]
+        public void ReadGroupCommand()
+        {
+            _protocol.ReadGroupCommand(_writtenGroupCommand);
+        }
+
+        [Benchmark]
+        public void ReadInvocationNoExclusions()
+        {
+            _protocol.ReadInvocation(_writtenInvocationNoExclusions);
+        }
+
+        [Benchmark]
+        public void ReadInvocationSmallExclusions()
+        {
+            _protocol.ReadInvocation(_writtenInvocationSmallExclusions);
+        }
+
+        [Benchmark]
+        public void ReadInvocationLargeExclusions()
+        {
+            _protocol.ReadInvocation(_writtenInvocationLargeExclusions);
+        }
+
+        private static IReadOnlyList<string> GenerateIds(int count)
+        {
+            var ids = new string[count];
+            for(var i = 0; i < count; i++)
+            {
+                ids[i] = Guid.NewGuid().ToString("N");
+            }
+            return ids;
+        }
+
+        private class DummyProtocol: IHubProtocol
+        {
+            private static byte[] _fixedOutput = new byte[] { 0x68, 0x68, 0x6C, 0x6C, 0x6F };
+
+            public string Name { get; }
+
+            public int Version => 1;
+
+            public TransferFormat TransferFormat => TransferFormat.Text;
+
+            public DummyProtocol(string name)
+            {
+                Name = name;
+            }
+
+            public bool IsVersionSupported(int version) => true;
+
+            public bool TryParseMessage(ref ReadOnlySequence<byte> input, IInvocationBinder binder, out HubMessage message)
+            {
+                throw new NotSupportedException();
+            }
+
+            public void WriteMessage(HubMessage message, IBufferWriter<byte> output)
+            {
+                output.Write(_fixedOutput);
```

Comment:
We could use real protocols too, but I was thinking it would be better to isolate the performance to the overhead of RedisProtocol specifically and use a HubProtocol that was consistent in its behavior.